### ISSUE TITLE
Need automatically convert responseText to JSON if appropriate header supplied

### ISSUE THREAD

**Issue by skotchio** [mailru/FileAPI#286]
In `complete` callback we need to get JSON parsed response if we set `Accept: 'application/json'` header for example:

    FileAPI.upload({
      url: 'upload.php',
      headers: {'Accept': 'application/json'},
      files: { file: file },
      complete: function (err, xhr) {
           var JSON = xhr.response;
      }});

**Comment by RubaXa** [comment]
Not quite, it is necessary to process the server response headers, but is not possible when using `Flash` or `IFrame`.

I can add the option `dataType: string`

**Comment by skotchio** [comment]
yes, will looks like as jQuery.ajax do. Thanks.
